### PR TITLE
Synthetic Kit printer (first of many medical gadgets to come)

### PR DESCRIPTION
**Pull request by fernandos13**
## About The Pull Request
this PR adds in the Synthetic trauma kit
with its gel system for fine tuning for this AND for the tools to come
its a exclusive item of corpsman. you can only get ONE printer. 

S.T.K   (Synthetic trauma kit)

is a new tool for our corpsman that high command is happy to announce
its got two modes that you can switch between using the lever in the tool
for either healing brute. OR burn damage
its got a transparent glass and a indicator so you can see the gel loaded in it move. and which mode your on

it uses our new GELL system a new system we are implanting all over our colonies too. its cells that use a medical gel "ITS NOT EDIBLE", that you can get at ANY powered nanomed(only comes with one) , or marinemed(16 to supply all corpsman), once your cell run out you can just connect it with a vendor that will fill it up ( no need to restock em manually)

we got at the moment two variants
the Gell classic comes with 840 gel units in it (easy to divide by most numbers), it can dispense 15 heals (only way to get is when you dispense the first STK for each corpsman)
the Gell tiny comes with 280 gel units in it can dispense 5 heals

both got a very modern gauge that will blink in different colors and patterns telling you how much you got left. and a transparent visor from which you can see how many you got left. 

## Why It's Good For The Game
gives a nifty tool for corpsman to use and opens up the space for furthering in more tools that will only impact the medic in his niche. without messing with other roles.

## Changelog
:cl:
add: Geltool STK for corpsman (trauma/burn kit on toggle uses gell)
add: Gell (currency for new geltools)
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
> its cells that use a medical gel "ITS NOT EDIBLE"

Let marines try to eat it. Give them organ damage for it.

File: code/game/objects/items/tools/medical_tools.dm [hunk 1]
```diff
@@ -11,14 +11,29 @@
...
+
+	user.visible_message("[user] sucks on \the [src].",
+	"You suck on \the [src].")
+	gelquantity -= 28 //let them do it till the puke
+
+	M.adjustFireLoss(10) // add a little bit of Burn damage
+	M.adjustToxLoss(5) // and tox for good measure
+	M.disabilities |= MUTE
+	addtimer(VARSET_CALLBACK(M, disabilities, M.disabilities & ~MUTE), 30 SECONDS)
+	to_chat(user, "you feel a acid taste in your mouth, as your lips tighten up</span>")
```
Comment (neohenry) [inline code/game/objects/items/tools/medical_tools.dm:35]: Crammatical corrections. Would be 

> **Y**ou

> a**n** acid**ic**

> in your mouth as you lips tighen up**.**

New sentence will be

> You feel an acidic taste on you mouth as your lips tighten up.

Comment (fernandos13) [reply code/game/objects/items/tools/medical_tools.dm:37]: thanks immensly 
i'll add this

File: code/game/objects/items/tools/medical_tools.dm [hunk 2]
```diff
@@ -0,0 +1,245 @@
+
+//CELLS here
+/obj/item/tool/geltool/gell  //Gell a CELL but GELL
+	name = "Gell classic"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:4]:
```suggestion
	name = "Gell classic vial"
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	desc = "A vial used to transport and store medical gel, NOT EDIBLE."
+	icon_state = "gell"
+	item_state = "gell"
+	icon = 'icons/obj/items/geltools.dmi'
+	var/gelcapacity = 840   //how much gel can you fit in this tool, THIS NUMBER WAS CHOSEN FOR ITS DIVISIBILITY
+	var/gelquantity = 840   //how much gel does it starts with
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/tool/geltool/gell/small
+	name = "Gell tiny"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:14]:
```suggestion
	name = "Gell tiny vial"
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+	"You suck on \the [src].")
+	gelquantity -= 35 // 8 sucks should suffice them without killing them
+
+	playsound(loc, 'sound/voice/alien_drool2.ogg', 30, 1)
+	playsound(loc, 'sound/voice/skeleton_warcry.ogg', 10,1,1)
+	M.adjustFireLoss(10) // add a little bit of Burn damage
+	M.adjustToxLoss(5) // and tox for good measure
+	M.disabilities |= MUTE
+	addtimer(VARSET_CALLBACK(M, disabilities, M.disabilities & ~MUTE), 30 SECONDS)
+	to_chat(user, "You feel an acidic taste on you mouth as your lips tighten up.</span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:37]:
```suggestion
	to_chat(user, "<span class='warning'>You feel an acidic taste on you mouth as your lips tighten up!</span>")
```
Is <span> even used?

File: code/game/objects/items/tools/medical_tools.dm [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+	update_icon()
+
+/obj/item/tool/geltool/gell/examine(mob/user)
+	..()
+	to_chat(user, "the gel gauge measures. <b>Gel quantity: [gelquantity]/[gelcapacity]</b>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:42]:
```suggestion
	to_chat(user, "<b>The gel gauge measures: [gelquantity]/[gelcapacity]</b>")
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+	gelquantity = gelcapacity
+	playsound(loc, 'sound/machines/hydraulics_1.ogg', 5, 1)
+	playsound(loc, 'sound/machines/switch.ogg', 50, 1)
+	to_chat(user, "<span class='warning'>You refill the [src]!</span>")
+	update_icon()
+
+//define for its TOOLs
+/obj/item/tool/geltool/gadget
+	name = "Medical gel based tool"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:66]:
```suggestion
	name = "medical gel-based tool"
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+	..()
+	if(!cell || cell.gelquantity <= 0)
+		item_state = "synthkitoff"
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+/obj/item/tool/geltool/gadget/proc/gel_use(M = 1)  //change the M if you want to further modify the efficient of a specific procedure
+	cell.gelquantity -= gelefficiency * M
+	usr.visible_message(,"<span class='notice'>the [src] used up [gelefficiency * M] gel in this procedure [cell.gelquantity] left.</span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:92]:
```suggestion
	usr.visible_message(,"<span class='notice'>The [src] used up [gelefficiency * M] gel in this procedure. [cell.gelquantity] left.</span>")
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+/obj/item/tool/geltool/gell/afterattack(obj/item/I, mob/user)
+	. = ..()
+	if(!istype(I, /obj/machinery/vending/nanomed || /obj/machinery/vending/MarineMed))
+		return
+
+	gelquantity = gelcapacity
+	playsound(loc, 'sound/machines/hydraulics_1.ogg', 5, 1)
+	playsound(loc, 'sound/machines/switch.ogg', 50, 1)
+	to_chat(user, "<span class='warning'>You refill the [src]!</span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:61]:
```suggestion
	to_chat(user, "<span class='notice'>You refill the [src]!</span>")
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+	if(user.get_inactive_held_item() != src)
+		return ..()
+	if(!cell)
+		return ..()
+	cell.update_icon()
+	user.put_in_active_hand(cell)
+	cell = null
+	playsound(user, 'sound/machines/click.ogg', 25, 1, 5)
+	playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+	to_chat(user, "<span class='notice'>You remove the Gell from [src].</span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:104]:
```suggestion
	to_chat(user, "<span class='notice'>You remove the Gell vial from [src].</span>")
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+	return TRUE
+
+/obj/item/tool/geltool/gadget/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/tool/geltool/gell))
+		if(!user.drop_held_item())
+			return
+		I.forceMove(src)
+		var/replace_install = "You replace the gell in [src]"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:115]:
```suggestion
		var/replace_install = "You replace the gell vial in [src]."
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+		if(!cell)
+			replace_install = "You install a gell in [src]"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:117]:
```suggestion
			replace_install = "You install a Gell vial in [src]."
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+			user.put_in_hands(cell)
+		cell = I
+		to_chat(user, "<span class='notice'>[replace_install] <b>Gel quantity: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		playsound(user, 'sound/weapons/guns/interact/rifle_reload.ogg', 25, 1, 5)
+		playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+		update_icon()
+
+// TOOLS down here
+/obj/item/tool/geltool/gadget/kitprinter
+	name = "Synthetic trauma kit"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:129]:
```suggestion
	name = "synthetic trauma kit"
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+	desc = "A tool that utilizes gel. to act as brute or a burn advanced kit. it have a level to toggle its mode"
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:131]:
```suggestion
	desc = "A tool that utilizes gel to act as brute or a burn advanced kit. It has a button to toggle between modes."
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+	update_icon()
+
+/obj/item/tool/geltool/gadget/kitprinter/examine(mob/user)
+	..()
+	if(cell)
+		to_chat(user, "the gel gauge measures. <b>Gel quantity: [cell.gelquantity]/[cell.gelcapacity]</b>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:162]:
```suggestion
		to_chat(user, "<b>The gel gauge measures: [gelquantity]/[gelcapacity]</b>")
```

File: code/game/objects/items/tools/medical_tools.dm (hunk 13 above)
Comment (fernandos13) [reply code/game/objects/items/tools/medical_tools.dm:131]: its actually a lever. in the sprite too. i see the "level" thing but lemme type it right

File: code/game/objects/items/tools/medical_tools.dm [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+	toggle(user)
+
+/obj/item/tool/geltool/gadget/kitprinter/proc/toggle(mob/user, silent)
+	if(brute)
+		brute = FALSE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 5)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 15)  //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] burn mode.</span>",
+		"<span class='notice'>You switch [src] lever to Burn.<b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:176]:
```suggestion
		"<span class='notice'>You switch [src] lever to burn mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+		update_kitprinter()
+		return
+
+	if(!brute)
+		brute = TRUE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 15)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 5 , 1) //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] burn mode.</span>",
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:185]:
```suggestion
			user.visible_message("<span class='notice'>[user] turns [src] to brute mode.</span>",
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+/obj/item/tool/geltool/gadget/kitprinter/attack_self(mob/user)
+	toggle(user)
+
+/obj/item/tool/geltool/gadget/kitprinter/proc/toggle(mob/user, silent)
+	if(brute)
+		brute = FALSE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 5)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 15)  //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] burn mode.</span>",
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:175]:
```suggestion
			user.visible_message("<span class='notice'>[user] turns [src] to burn mode.</span>",
```

File: code/game/objects/items/tools/medical_tools.dm [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+		update_kitprinter()
+		return
+
+	if(!brute)
+		brute = TRUE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 15)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 5 , 1) //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] burn mode.</span>",
+		"<span class='notice'>You switch [src] lever to Brute.<b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
```
Comment (XSlayer300) [inline code/game/objects/items/tools/medical_tools.dm:186]:
```suggestion
		"<span class='notice'>You switch [src] lever to brute mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
```

File: code/game/objects/items/tools/medical_tools.dm (hunk 14 above)
Comment (fernandos13) [reply code/game/objects/items/tools/medical_tools.dm]: cell is a var here i cannot remove it. 
i made the change to the rest tho  "<b>The gel gauge measures: "

**Comment by Walarks** [comment]
This is just my opinion and maybe people would like higher healing for marines, but if this only increases the healing corpsmen can dish out, they should get their healing nerfed in a different way as they can already heal marines quickly.

**Comment by TiviPlus** [comment]
Balance freeze

**Pull request closed without merging**